### PR TITLE
Store method to execute array of operations in a single SQL transaction

### PR DESCRIPTION
**Pull request by shrishrirang**
- The executeBatch methods supports upserting of a record or deleting a record using its ID. 
- This is helpful for performing multiple operations in a single batch.
- executeBatch can be extended to support more operation types as and when needed.

### CODE REVIEW

**Comment by shrishrirang** [comment]
\+ @danderson00

File: sdk/src/Platforms/cordova/MobileServiceSqliteStore.js [hunk 1]
```diff
@@ -222,17 +236,14 @@ var MobileServiceSqliteStore = function (dbName) {
...
         }
 
-        this._db.transaction(function (transaction) {
-            // Execute the INSERT and UPDATE statements.
-            for (var i = 0; i < statements.length; i++) {
-                transaction.executeSql(statements[i], parameters[i]);
+        // Execute the INSERT and UPDATE statements.
+        for (i = 0; i < statements.length; i++) {
+            if (this._editStatement) { // test hook
+                statements[i] = this._editStatement(statements[i]);
```
Comment (danderson00) [inline sdk/src/Platforms/cordova/MobileServiceSqliteStore.js]: Nasty...

**Comment by danderson00** [comment]
Let's have a quick chat when you get in.

**Comment by shrishrirang** [comment]
Discussed all comments with @danderson00. Merging this PR now.

**Pull request merged**